cli/pkg/autodetect/kind.go

```diff
@@ -48,18 +42,23 @@ type ClusterDetails struct {
 	K8SVersion *version.Version
 }
 
-var (
-	ErrCannotDetectK8sVersion  = fmt.Errorf("cannot detect k8s version")
-	ErrCannotDetectClusterKind = fmt.Errorf("cannot detect cluster kind")
-)
+func GetK8SVersion() *version.Version {
+	return currentClusterDetails.K8SVersion
+}
 
-func DetectK8SClusterDetails(ctx context.Context, kc string, client *kube.Client) (ClusterDetails, error) {
+func GetClusterKind() Kind {
+	return currentClusterDetails.Kind
+}
+
+func SetK8SClusterDetails(ctx context.Context, kc string, client *kube.Client)  {
```

Comment:
Perhaps a small comment, mentioning that this function will set global variables, and is a requirement for the `GetK8SVersion` and `GetClusterKind` above to work.